third_party/terraform/resources/resource_logging_sink.go

```diff
@@ -56,21 +73,39 @@ func flattenResourceLoggingSink(d *schema.ResourceData, sink *logging.LogSink) {
 	d.Set("writer_identity", sink.WriterIdentity)
 }
 
-func expandResourceLoggingSinkForUpdate(d *schema.ResourceData) *logging.LogSink {
+func expandResourceLoggingSinkForUpdate(d *schema.ResourceData) (*logging.LogSink, string) {
 	// Can only update destination/filter right now. Despite the method below using 'Patch', the API requires both
 	// destination and filter (even if unchanged).
 	sink := logging.LogSink{
-		Destination: d.Get("destination").(string),
-		Filter:      d.Get("filter").(string),
+		Destination:     d.Get("destination").(string),
+		Filter:          d.Get("filter").(string),
+		ForceSendFields: []string{"Destination", "Filter"},
 	}
 
+	updateFields := []string{}
 	if d.HasChange("destination") {
-		sink.ForceSendFields = append(sink.ForceSendFields, "Destination")
+		updateFields = append(updateFields, "destination")
 	}
 	if d.HasChange("filter") {
-		sink.ForceSendFields = append(sink.ForceSendFields, "Filter")
+		updateFields = append(updateFields, "filter")
+	}
+	if d.HasChange("bigquery_options") {
+		sink.BigqueryOptions = expandLoggingSinkBigqueryOptions(d.Get("bigquery_options"))
+		updateFields = append(updateFields, "bigqueryOptions")
+	}
+	return &sink, strings.Join(updateFields, ",")
+}
+
+func expandLoggingSinkBigqueryOptions(v interface{}) *logging.BigQueryOptions {
+	if v == nil {
+		return nil
+	}
+	options := v.([]interface{})[0].(map[string]interface{})
```

Comment:
You are casting into an interface slice but not checking if the slice is empty or if it is non-empty but contains a nil instead on another interface. This will be a panic if you encounter any of these situations.
[This is an example](https://github.com/terraform-providers/terraform-provider-google/blob/master/google/resource_container_cluster.go#L1711) of guarding against these conditions.

Reply:
Oh I didn't know there's chance v become empty slice or likewise. I was expecting Terraform SDK handles this to make it always be `nil` or `[]map[string]interface{}`...
Fixed in 63bc19c